### PR TITLE
Add some comments to test_example_qir_phi_int.ll

### PR DESCRIPTION
**Pull request by doug-q**

### CODE REVIEW

**Comment by doug-q** [comment]
@cqc-melf Please review (note this is based on example/phiregister)

File: tests/qir/test_example_qir_phi_int.ll [hunk 1]
```diff
@@ -14,39 +14,47 @@ entry:
...
+then:                                            ; preds = %entry
   %10 = zext i2 %5 to i7
   %11 = sub i7 125, %10
-  %12 = trunc i7 %11 to i5 
-  br label %continue3
+  %12 = trunc i7 %11 to i5 ; b = c - a
+
+  ; a = %5
+  ; b = %12
+  ; c = 125
```
Comment (cqc-melf) [inline tests/qir/test_example_qir_phi_int.ll:52]: I think I missed this yesterday, there should be two conditional branches in here?
One based on a[0] and one based on a[1]?
From my understanding there should be one additional branch based on a[0] setting c to c - b

File: tests/qir/test_example_qir_phi_int.ll [hunk 2]
```diff
@@ -14,39 +14,47 @@ entry:
...
 
-then:                                             ; preds = %entry
-  br label %continue
-
-continue:                                         ; preds = %entry, %then
   %7 = and i2 2, %5
   %8 = lshr i2 %7, 1
   %9 = trunc i2 %8 to i1
-  br i1 %9, label %then1, label %continue3
+  br i1 %9, label %then, label %continue; if a[2]
```
Comment (cqc-melf) [inline tests/qir/test_example_qir_phi_int.ll]: a has only size 2, so a[2] is not defined?

**Review by cqc-melf — COMMENTED** [review]
If my comments are unclear, feel free to ask

File: tests/qir/test_example_qir_phi_int.ll (hunk 1 above)
Comment (doug-q) [reply tests/qir/test_example_qir_phi_int.ll:52]: This is true, but `b == 0`, so I optimised it out. I'll put it back

File: tests/qir/test_example_qir_phi_int.ll (hunk 2 above)
Comment (doug-q) [reply tests/qir/test_example_qir_phi_int.ll]: yes this an off by one, will fix.

File: tests/qir/test_example_qir_phi_int.ll (hunk 1 above)
Comment (cqc-melf) [reply tests/qir/test_example_qir_phi_int.ll:52]: The conditional is based on the result of the quantum measurement, it should not be possible to optimise this? Or did I missed something?

Comment (doug-q) [reply tests/qir/test_example_qir_phi_int.ll:52]: c is 125, b = 0. Then, if a[0] set c = c - b = c. One can drop the redundant write.

**Review by cqc-melf — APPROVED** [review]

**Comment by cqc-melf** [comment]
Feel free to merge this now

**Pull request merged**